content/articles/email-validator-in-flutter/index.md

```diff
@@ -0,0 +1,238 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /flutter-email-validator/
+title: Flutter Email Validator
+description: This article will provide a guide on how to use Email Validator package in flutter to validate emails during user input.
+author: joni-kano
+date: 2021-10-26T00:00:00-11:27
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/flutter-email-validator/hero.png
+    alt: Flutter Email Validator Hero Image
+---
+Email validator is a Flutter package that checks whether an email is valid or not. It is not based on regular expression, but rather checks if the email entered is linked to a reliable domain. Domains that exist are Gmail, Yahoo, Outlook, among others.
+<!--more-->
+### Key takeaways
+- What is an email validator?
+- Explaining how an email validator works.
+- Create a simple app to show email validator.
+- Understand the implementation of email validator.
+
+### Working mechanism of email validator
+Email validator runs a check on the emails through a process called `swift process`. This process detects typographical errors or deliberate mistakes. Furthermore, it helps to organize and clean all email addresses in a flutter application.
+
+For this reason, it protects the email sender score while maximizing the efficiency of the flutter application. The key concepts from the use of this package are validity, risk, and the invalidity of any email address.
+
+This package consists of a class and a function. 
+
+The constructor for this class is pretty simple: Create a new instance of this class and set the validator type to either 1 or 2 (defaults to 1). The constructor should pass a string and an array of strings for validation.The arguments are also key factors.
+
+The first argument is self-explanatory. It is just a boolean that specifies whether we are using the `local-part@domain.com` format or `domain.com` as our string to validate.
+
+The second argument allows you to pass an array of strings that are valid domains for the email. This argument is only necessary if the string passed in as our first argument is in `domain.com` format; otherwise, it can be omitted.
+
+If either of the first two arguments contains an error, then false should be returned. Otherwise, true should be returned, and our string should be validated.
```

Comment:
Where are the arguments? Explain them in detail with a proper example to enable understanding of the concept.